File: core/evidence/inference_integrity_engine.py
from __future__ import annotations

from typing import Any, Dict, List, Optional


def model_inference_integrity(
    evidence: List[str],
    supporting: List[str],
    contradicting: List[str],
    fragility: Dict[str, Any],
    missing_evidence: Optional[List[str]] = None,
) -> Dict[str, Any]:
    caps = fragility.get("confidence_limits", {})
    return {
        "basis": fragility.get("basis", {}),
        "supporting_evidence": sorted(set(supporting or evidence)),
        "contradicting_evidence": sorted(set(contradicting or [])),
        "missing_evidence": sorted(set(missing_evidence or fragility.get("missing_support", []))),
        "uncertainty_factors": fragility.get("contradiction_pressure", []),
        "confidence_caps": [caps.get("max_score", 1.0)],
        "unsupported_dimensions": fragility.get("missing_support", []),
        "fragility": fragility,
    }

File: src/evidence/inferenceIntegrityEngine.ts
/**
 * Converted from Python: core/evidence/inference_integrity_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function modelInferenceIntegrity(evidence: any, supporting: any, contradicting: any, fragility: any, missing_evidence: any = null): any {
  var caps: any = py.get(fragility, "confidence_limits", {});
  return {"basis": py.get(fragility, "basis", {}), "supporting_evidence": py.sorted(py.toSet(py.or2(supporting, () => (evidence)))), "contradicting_evidence": py.sorted(py.toSet(py.or2(contradicting, () => ([])))), "missing_evidence": py.sorted(py.toSet(py.or2(missing_evidence, () => (py.get(fragility, "missing_support", []))))), "uncertainty_factors": py.get(fragility, "contradiction_pressure", []), "confidence_caps": [py.get(caps, "max_score", py.F(1.0))], "unsupported_dimensions": py.get(fragility, "missing_support", []), "fragility": fragility};
}
